lib/scihist_digicoll/task_helpers/subject_creator_adjuster.rb

```diff
@@ -0,0 +1,40 @@
+module ScihistDigicoll
+  module TaskHelpers
+    class SubjectCreatorAdjuster
+      attr_accessor :metadata_map, :keys_to_check_for, :changes, :errors
+      
+      def initialize
+        @changes = []
+        @errors = []
+        @metadata_map = {
+          "Bredig, Georg, 1868-" => "Bredig, Georg, 1868-1944",
+          "Caruso, David J."     => "Caruso, David J., (David Joseph), 1978-"
+        }
+        @keys_to_check_for = @metadata_map.keys
+      end
+
+      def process_work(work)
+        update_subject(work) unless (work.subject              & @keys_to_check_for).empty?
+        update_creator(work) unless (work.creator.map(&:value) & @keys_to_check_for).empty?
+      end
+
+      def update_subject(work)
+        updated_subject = work.subject.filter_map { |s| @metadata_map.fetch(s, s) }
```

Comment:
I don't understand what this `filter_map` is doing. Why `filter_map` instead of just `map` (aka `collect`)?  It makes me think it's going to be removing subjects... but it's not intended to, is it? 

Should there be a conditional in here, like only process and `save` the work if it includes a subject we're interested in?  

Or the fancier/quicker way to do this would be only fetch works we're interested in in the first place, but maybe that's not necessary -- definitely not for a one-off, we can do more performance opt later if we end up re-using this maybe.

Reply:
The `filter_map` allows you to specify a null value for a key in `@metadata_map` which does allow you to remove subjects. But I'm glad you mention it because that doesn't work at all with creators! Will simplify this before we run it. Good catch.

Reply:
I guess filter_map would let you have a Hash with a `nil` valuae meaning "delete this subject"? Is that intended? Should be documented. And come to think of it, if we plan on re-using this thing for possible future data changes (which I think seems not unlikely), there should be maybe some tests of this class? Not exhaustive tests, just like one or two or three simple tests.

Reply:
@eddierubeiz the filter_map is still there?  If we want to support `nil` value, it should be both documented and tested... otherwise remove the filter_map?

I still have the same question as above:

> Should there be a conditional in here, like only process and save the work if it includes a subject we're interested in?

> Or the fancier/quicker way to do this would be only fetch works we're interested in in the first place, but maybe that's not necessary -- definitely not for a one-off, we can do more performance opt later if we end up re-using this maybe.

Reply:
Yeah, there's a conditional there of course -- only relevant works are saved. I'll put in a comment, though, to make it clearer. Sorry I missed your question!

Reply:
okay, sorry for missing that, nice! just filter_map then!

Reply:
Got rid of it!